### PR TITLE
Opal 680

### PR DESCRIPTION
**Pull request by CK-USAF**

### CODE REVIEW

File: .github/workflows/validate_code.yml [hunk 1]
```diff
@@ -70,4 +70,5 @@ jobs:
         export AWS_ACCESS_KEY_ID=minioadmin
         export AWS_SECRET_ACCESS_KEY=minioadmin
         export FSSPEC_S3_ENDPOINT_URL=http://172.17.0.1:9000/
+        export LOGGER_LEVEL=DEBUG
```
Comment (AM-USAF) [inline .github/workflows/validate_code.yml:73]: Do we need to export the logger level, there is no default if we don't?

File: task_queue/cli/work_queue_service_cli.py [hunk 2]
```diff
@@ -208,13 +210,15 @@ def main(periodic_functions, work_queue, period_sec=10):
...
     )
+    logger.setLevel(settings.logger_level)
     settings.log_settings()
 
     # Check if dependent arguments were provided
     # This validation could be reworked into the settings model
     success, error_string = validate_args(settings.model_dump())
 
     if not success:
+        logger.error(error_string)
```
Comment (AM-USAF) [inline task_queue/cli/work_queue_service_cli.py:221]: You have some instances where you log errors and some where you don't. Were you just picking errors you thought were important or did you accidentally miss them?

File: task_queue/logger.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    file_handler = logging.FileHandler(get_log_fp())
+    stream_handler = logging.StreamHandler(sys.stdout)
+    # Add formatting to the statements
+    fmt = logging.Formatter(fmt="%(asctime)s [%(levelname)s]: %(message)s")
+    stream_handler.setFormatter(fmt)
+    file_handler.setFormatter(fmt)
+    # Add handlers to logger
+    _logger.addHandler(stream_handler)
+    _logger.addHandler(file_handler)
+    # set log level
```
Comment (AM-USAF) [inline task_queue/logger.py:32]: Capital S

Comment (CK-USAF) [reply task_queue/logger.py]: changed

File: .github/workflows/validate_code.yml (hunk 1 above)
Comment (CK-USAF) [reply .github/workflows/validate_code.yml]: I fixed the pydantic it didnt like the property i had default now works

File: tests/test_logger.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+
+def assert_log_fp(fp):
+    log_dir = os.getcwd() + '/log'
+    assert log_dir in fp
+    assert ".log" in fp
+    
+
+def test_log_dir_created():
+    new_log_fp = get_log_fp()
```
[the remark this answers is not in the document]
Comment (CK-USAF) [reply tests/test_logger.py]: added

File: task_queue/cli/work_queue_service_cli.py (hunk 2 above)
Comment (CK-USAF) [reply task_queue/cli/work_queue_service_cli.py:221]: I went back through and added logger statements to the exceptions I missed, I think I got them all

**Review by AM-USAF — APPROVED** [review]

**Pull request merged**